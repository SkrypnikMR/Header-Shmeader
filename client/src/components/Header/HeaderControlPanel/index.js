import { withRouter } from 'react-router';
import HeaderControlPanel from './HeaderControlPanel.jsx';
import { connect } from 'react-redux';
import { setValue } from '/src/Store/user/actions';
import { userThemeMode } from '/src/Store/user/selectors';

const mapStateToProps = state => ({ themeMode: userThemeMode(state) });
const mapDispatchToProps = dispatch => ({
    setValue: payload => dispatch(setValue(payload)),
});

export default connect(mapStateToProps, mapDispatchToProps)(withRouter(HeaderControlPanel));
