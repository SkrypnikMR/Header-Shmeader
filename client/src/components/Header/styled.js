import styled from 'styled-components';

export const StHeader = styled.div`
    display:flex;
    justify-content: flex-start;
    align-items: center;
    height:10vh;
    background-color: rgba(0, 0, 0, 0.4);
`;
