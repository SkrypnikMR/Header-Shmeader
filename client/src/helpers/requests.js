export const getRequest = async (url, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = { method: 'GET', headers };
  const answer = await fetch(url, options);
  return answer.json();
};
export const postRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'POST',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(url, options);
  return answer.json();
};

export const putRequest = async (url, body, optionalHeaders = {}) => {
  const headers = Object.assign({ 'Content-Type': 'application/json' }, optionalHeaders);
  const options = {
    method: 'PUT',
    headers,
    body: JSON.stringify(body),
  };
  const answer = await fetch(url, options);
  return answer.json();
};
export const createAuthHeader = (support) => { return { Authorization: support.getCookie('token') }; };
