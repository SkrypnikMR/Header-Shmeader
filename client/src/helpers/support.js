export const support = {
    setSessionStorageItem: (name, data) => {
        if (typeof data !== 'string') data = JSON.stringify(data);
        sessionStorage.setItem(name, data);
    },
    killSessionStorageItem: (name) => {
        sessionStorage.removeItem(name);
    },
    getSessionStorageItem: (name) => {
        const item = sessionStorage.getItem(name);
        return item?.charAt(0) === '{' ? JSON.parse(item) : item;
    },
    getPrettyTime: time => `${new Date(Number(time)).toTimeString().substr(0, 8)}`,
    getMessagesFolders: (rooms) => {
        const messagesFolders = {};
        rooms.forEach((room) => {
            const { room_name } = room;
            messagesFolders[room_name] = [];
        });
        return messagesFolders;
    },
    filteredRooms: (filterValue, rooms) => {
        const result = rooms.filter(room => room.room_name.includes(filterValue));
        return result.length === 0 ? null : result;
    },
};
