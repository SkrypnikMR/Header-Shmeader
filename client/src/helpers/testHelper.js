import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import { BrowserRouter as Router } from 'react-router-dom';

export const shallowSmart = (component, store) => {
    const core = store
        ? <Provider store={store}>{component}</Provider>
        : component;
    const test = <Router>{core}</Router>;
    return shallow(test);
};
