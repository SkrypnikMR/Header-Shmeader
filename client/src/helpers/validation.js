import { regExp } from '../constants/regExp';

export const validation = {
    registrationValidation: ({ email, password, confirm, firstName, lastName }) => {
        let answer = { message: '', isValid: true };

        if (!regExp.emailRegExp.test(email) || !email) answer = { message: 'Invalid email', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) answer = { message: 'Invalid password', isValid: false };
        if (!regExp.firstNameRegExp.test(firstName) || !firstName) answer = { message: 'Invalid firstName', isValid: false };
        if (!regExp.lastNameRegExp.test(lastName) || !lastName) answer = { message: 'Invalid lastName', isValid: false };
        if (password !== confirm) answer = { message: 'Password mismatch', isValid: false };
        return answer;

    },
    loginValidation: ({ email, password }) => {
        let answer = { message: '', isValid: true };

        if (!regExp.emailRegExp.test(email) || !email) answer = { message: 'Invalid email', isValid: false };
        if (!regExp.passwordRegExp.test(password) || !password) answer = { message: 'Invalid password', isValid: false };
        return answer;
    },
};
