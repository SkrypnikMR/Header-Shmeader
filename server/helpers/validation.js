const regExp = require('../Const/regExp')

module.exports = {
    registrationValidation: (req, res, next) => {
        const { email, password, firstName, lastName } = req.body;
        let answer = { message: '', isValid: true };

        if (!regExp.emailRegExp.test(email) || !email) {
            answer = { message: 'Invalid email', isValid: false };
        }
        if (!regExp.passwordRegExp.test(password) || !password) {
            answer = { message: 'Invalid password', isValid: false };
        }
        if (!regExp.firstNameRegExp.test(firstName) || !firstName) {
            answer = { message: 'Invalid firstName', isValid: false };
        }
        if (!regExp.lastNameRegExp.test(lastName) || !lastName) {
            answer = { message: 'Invalid lastName', isValid: false };
        }
        req.validation = {message: answer.message, isValid: answer.isValid}
        next();
    },
    loginValidation: (req, res, next) => {
        const { email, password } = req.body;
        let answer = { message: '', isValid: true };

        if (!regExp.emailRegExp.test(email) || !email) {
            answer = { message: 'Invalid email', isValid: false };
        }
        if (!regExp.passwordRegExp.test(password) || !password) {
            answer = { message: 'Invalid password', isValid: false };
        }
        req.validation = {message: answer.message, isValid: answer.isValid}
        next();
    },
};
