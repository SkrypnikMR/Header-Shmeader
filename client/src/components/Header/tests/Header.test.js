import React from 'react';
import { shallowSmart, mountSmart } from '/src/helpers/testHelper';
import Header from '../Header';

describe('Header', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<Header />);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render p', () => {
        const component = mountSmart(<Header />);
        expect(component.find('Logo')).toHaveLength(1);
    });
    it('should render inputs', () => {
        const component = mountSmart(<Header />);
        expect(component.find('HeaderControlPanel')).toHaveLength(1);
    });
});
