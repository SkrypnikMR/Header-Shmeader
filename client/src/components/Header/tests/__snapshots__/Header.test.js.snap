// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Should match snapshot 1`] = `"<div class=\\"styled__StHeader-zzrvsn-0 hDaupM\\"><div class=\\"styled__StLogo-sc-9vy3sh-0 krciQz\\"><img src=\\"./public/assets/images/logo.png\\"/><h1>logo</h1></div><div class=\\"styled__StControl-sc-1i99xqk-0 joBAxQ\\"><button id=\\"theme_btn\\" type=\\"button\\" value=\\"theme_btn\\" color=\\"white\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 iwKgks\\">ðŸŒ—</button><button id=\\"en\\" type=\\"button\\" value=\\"en\\" color=\\"white\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 iwKgks\\">english_translate_btn</button><button id=\\"ru\\" type=\\"button\\" value=\\"ru\\" color=\\"white\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 iwKgks\\">russian_translate_btn</button><button id=\\"ar\\" type=\\"button\\" value=\\"ar\\" color=\\"white\\" height=\\"10vh\\" width=\\"60px\\" class=\\"styled__StButton-sc-1a6hck8-0 iwKgks\\">arabian_translate_btn</button></div></div>"`;
